samples/amber.cc

```diff
@@ -168,7 +198,11 @@ bool ParseArgs(const std::vector<std::string>& args, Options* opts) {
         return false;
       }
 
-      int32_t val = std::stoi(std::string(args[i]));
+      int32_t val = 0;
+      if (1 != ParseOneInt(args[i].c_str(), &val)) {
```

Comment:
Can you flip these conditions, we usually put the `!= 1` last.

Reply:
Actually, can we move the comparison into the function and have it return bool? The 1/2 from these is ... confusing